python/cudf_polars/cudf_polars/dsl/expressions/struct.py

```diff
@@ -0,0 +1,134 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024-2025, NVIDIA CORPORATION & AFFILIATES.
+# SPDX-License-Identifier: Apache-2.0
+# TODO: remove need for this
+# ruff: noqa: D101
+"""Struct DSL nodes."""
+
+from __future__ import annotations
+
+from enum import IntEnum, auto
+from io import StringIO
+from typing import TYPE_CHECKING, Any, ClassVar
+
+import pylibcudf as plc
+
+from cudf_polars.containers import Column
+from cudf_polars.dsl.expressions.base import ExecutionContext, Expr
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from polars.polars import _expr_nodes as pl_expr
+
+    from cudf_polars.containers import DataFrame, DataType
+
+__all__ = ["StructFunction"]
+
+
+class StructFunction(Expr):
+    class Name(IntEnum):
+        """Internal and picklable representation of polars' `StructFunction`."""
+
+        FieldByIndex = auto()
+        FieldByName = auto()
+        RenameFields = auto()
+        PrefixFields = auto()
+        SuffixFields = auto()
+        JsonEncode = auto()
+        WithFields = auto()  # TODO: Implement in Polars
+        MapFieldNames = auto()  # TODO: Implement in Polars
+        MultipleFields = auto()  # TODO: Implement in Polars
+
+        @classmethod
+        def from_polars(cls, obj: pl_expr.StructFunction) -> Self:
+            """Convert from polars' `StructFunction`."""
+            try:
+                function, name = str(obj).split(".", maxsplit=1)
+            except ValueError:  # pragma: no cover
```

Comment:
I see a similar pattern in `BooleanFunction.Name.from_polars`, but without the no-cover pragmas. Are we able to reach this somehow?